src/libraries/System.Net.Http/src/System/Net/Http/SocketsHttpHandler/HttpConnectionPool.cs

```diff
@@ -519,21 +516,34 @@ public byte[] Http2AltSvcOriginUri
             Socket? socket = null;
             SslStream? sslStream = null;
             TransportContext? transportContext = null;
+            Http2Connection[]? lastHttp2Connections = _http2Connections;
```

Comment:
Shouldn't we capture `lastHttp2Connections` before/in `GetHttp2ConnectionAcceptingNewStreams`? Otherwise, by the time we get here, a new connection with free slots could be captured. Later, when `ReferenceEquals(currentHttp2Connections, lastHttp2Connections)` wrongly indicates the _http2Connections hasn't changed, it will skip trying to `FindHttp2ConnectionWithFreeStreamSlot` with the `_http2ConnectionCreateLock` even though there might be one, and call `AddHttp2Connection` unnecessarily.

Reply:
Good catch. I already fixed it.